File: src/prepare_environment_without_goalkeeper.py
#! /usr/bin/env python3

# import usefull packages
import rospy
from turtlesim.srv import Spawn, SpawnRequest


class EnvironmentClass():

    def __init__(self):


        # wait for the service
        rospy.wait_for_service('/spawn')

        # create turtles for the goal
        #####################T1###################
        self.T1_service_call = rospy.ServiceProxy(
            '/spawn', Spawn)
        self.T1_object = SpawnRequest()

        self.T1_object.x = 2.5

        self.service_client_T1_result = self.T1_service_call(
            self.T1_object)
            
        #####################T2###################
        self.T2_service_call = rospy.ServiceProxy(
            '/spawn', Spawn)
        self.T2_object = SpawnRequest()

        self.T2_object.x = 3.5

        self.service_client_T2_result = self.T2_service_call(
            self.T2_object)
            
        #####################T3###################
        self.T3_service_call = rospy.ServiceProxy(
            '/spawn', Spawn)
        self.T3_object = SpawnRequest()

        self.T3_object.x = 7.5

        self.service_client_T3_result = self.T3_service_call(
            self.T3_object)
        
        #####################T4###################
        self.T4_service_call = rospy.ServiceProxy(
            '/spawn', Spawn)
        self.T4_object = SpawnRequest()

        self.T4_object.x = 8.5

        self.service_client_T4_result = self.T4_service_call(
            self.T4_object)
            
        #####################T5###################
        self.T5_service_call = rospy.ServiceProxy(
            '/spawn', Spawn)
        self.T5_object = SpawnRequest()

        self.T5_object.x = 9.5

        self.service_client_T5_result = self.T5_service_call(
            self.T5_object)
        
        #####################T6###################
        self.T6_service_call = rospy.ServiceProxy(
            '/spawn', Spawn)
        self.T6_object = SpawnRequest()

        self.T6_object.x = 1.5

        self.service_client_T6_result = self.T6_service_call(
            self.T6_object)


        rospy.loginfo("Successfully build the goal")



if __name__ == '__main__':
    rospy.init_node('environment_node')
    track_object = EnvironmentClass()
    rospy.spin()

File: src/prepare_environment_without_goalkeeper.cpp
#include <ros/ros.h>
#include <turtlesim/Spawn.h>

class EnvironmentClass
{
    protected:
    
    // ROS Objects
    ros::NodeHandle nh_;
    
    //ROS Service
    ros::ServiceClient T1_service_call; 
    std::string srv_name_;
    turtlesim::Spawn T1_object;
    ros::ServiceClient T2_service_call;
    turtlesim::Spawn T2_object;
    ros::ServiceClient T3_service_call; 
    turtlesim::Spawn T3_object;
    ros::ServiceClient T4_service_call; 
    turtlesim::Spawn T4_object;
    ros::ServiceClient T5_service_call; 
    turtlesim::Spawn T5_object;
    ros::ServiceClient T6_service_call; 
    turtlesim::Spawn T6_object;
   
    
    public:
    	EnvironmentClass(std::string name): srv_name_(name)
    	{
    	
    	    ros::service::waitForService(name);
	    
	    ////////////////T1//////////////////
            T1_service_call = nh_.serviceClient<turtlesim::Spawn>(name);
            ros::Duration(0.1).sleep();
            T1_object.request.x = 1.5;
            T1_service_call.call(T1_object);
            
            ////////////////T2//////////////////
            T2_service_call = nh_.serviceClient<turtlesim::Spawn>(name);
            ros::Duration(0.1).sleep();
            T2_object.request.x = 2.5;
            T2_service_call.call(T2_object);
            
            ////////////////T3//////////////////
            T3_service_call = nh_.serviceClient<turtlesim::Spawn>(name);
            ros::Duration(0.1).sleep();
            T3_object.request.x = 3.5;
            T3_service_call.call(T3_object);
            
            ////////////////T4//////////////////
            T4_service_call = nh_.serviceClient<turtlesim::Spawn>(name);
            ros::Duration(0.1).sleep();
            T4_object.request.x = 7.5;
            T4_service_call.call(T4_object);
            
            ////////////////T5//////////////////
            T5_service_call = nh_.serviceClient<turtlesim::Spawn>(name);
            ros::Duration(0.1).sleep();
            T5_object.request.x = 8.5;
            T5_service_call.call(T5_object);
            
            ////////////////T6//////////////////
            T6_service_call = nh_.serviceClient<turtlesim::Spawn>(name);
            ros::Duration(0.1).sleep();
            T6_object.request.x = 9.5;
            T6_service_call.call(T6_object);
            
            ROS_INFO("Successfully build the goal");
            
        }
};
        

int main(int argc, char** argv)
{
  ros::init(argc, argv, "environment_node");
  EnvironmentClass track_object("/spawn");
  
  
  return 0;
}
